# Day 10 - Final Project Documentation

## Objective
Prepare the final documentation and validate all project components before submission.

Completed:
- Data Loading
- Data Cleaning
- EDA
- Financial Analysis
- Stock Analysis
- Sector Analysis
- SQL Analysis
- Database Validation

In [6]:
import sqlite3
import pandas as pd

conn = sqlite3.connect("../database/nifty100.db")

tables = pd.read_sql(
"""
SELECT name
FROM sqlite_master
WHERE type='table'
""", conn)

tables

,name
0,companies
1,analysis
2,balancesheet
3,cashflow
4,profitandloss
5,documents
6,prosandcons


In [7]:
for t in tables["name"]:
    df = pd.read_sql(f"SELECT COUNT(*) as Rows FROM {t}", conn)
    print(t)
    print(df)

companies
   Rows
0    93
analysis
   Rows
0    21
balancesheet
   Rows
0  1313
cashflow
   Rows
0  1188
profitandloss
   Rows
0  1277
documents
   Rows
0  1586
prosandcons
   Rows
0    17


In [8]:
import os

print(os.getcwd())

c:\Users\HP\OneDrive\Desktop\N100_Project\notebooks


In [9]:
import os

print(os.path.abspath("../database/nifty100.db"))
print(os.path.exists("../database/nifty100.db"))

c:\Users\HP\OneDrive\Desktop\N100_Project\database\nifty100.db
True


In [10]:
import sqlite3

conn = sqlite3.connect("../database/nifty100.db")

cursor = conn.cursor()

cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")

print(cursor.fetchall())

[('companies',), ('analysis',), ('balancesheet',), ('cashflow',), ('profitandloss',), ('documents',), ('prosandcons',)]


In [11]:
cursor = conn.cursor()
cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
print(cursor.fetchall())

[('companies',), ('analysis',), ('balancesheet',), ('cashflow',), ('profitandloss',), ('documents',), ('prosandcons',)]


In [12]:
tables = pd.read_sql("""
SELECT name
FROM sqlite_master
WHERE type='table'
""", conn)

tables

,name
0,companies
1,analysis
2,balancesheet
3,cashflow
4,profitandloss
5,documents
6,prosandcons


In [13]:
for t in tables["name"]:
    df = pd.read_sql(f"SELECT COUNT(*) as Rows FROM {t}", conn)
    print(f"\n===== {t} =====")
    print(df)


===== companies =====
   Rows
0    93

===== analysis =====
   Rows
0    21

===== balancesheet =====
   Rows
0  1313

===== cashflow =====
   Rows
0  1188

===== profitandloss =====
   Rows
0  1277

===== documents =====
   Rows
0  1586

===== prosandcons =====
   Rows
0    17


## Final Database Summary

| Table | Rows |
|--------|------|
| Companies | 93 |
| Analysis | 21 |
| Balance Sheet | 1313 |
| Cash Flow | 1188 |
| Profit & Loss | 1277 |
| Documents | 1586 |
| Pros and Cons | 17 |

## Technologies Used

- Python
- Pandas
- SQLite
- Jupyter Notebook
- VS Code
- Git
- GitHub

## Key Learnings

- Database Management using SQLite
- Data Validation
- SQL Queries
- Financial Data Analysis
- Git & GitHub Version Control

## Conclusion

The N100 Financial Intelligence Platform data was successfully validated and documented.
All core database tables were verified and are ready for further analytics and visualization.